marshmallow/fields.py

```diff
@@ -623,7 +623,10 @@ def _validated(self, value):
         if isinstance(value, uuid.UUID):
             return value
         try:
-            return uuid.UUID(value)
+            if isinstance(value, bytes):
```

Comment:
@JeffBerger I removed the length check here so that a `ValidationError` would be raised for input with improper length instead of an uncaught TypeError.